### DISCUSSION TITLE
Feedback Copilot x private beta

### DISCUSSION THREAD

**Discussion by JonasDev17** [community/community#60239] [Copilot Conversations] [answered]
### Select Topic Area

Product Feedback

### Body

I have been using GitHub Copilot x for a few hours now, but I'm not particularly satisfied with the code it generates.

One issue I've noticed is that it often forgets about code I've generated or written earlier. For example, it generated some code with nullable and optional parameters, even though I explicitly indicated that I didn't want them to be optional and that two of them should not be nullable. After I had it make the changes, I continued to interact with Copilot, but it reverted the non-nullable parameter back to being nullable. In my opinion, having access to the actual ChatGPT 4 would be more helpful and provide better assistance.

While I understand that I could have written the code myself, and probably faster since I knew exactly what I wanted, I was hoping to test Copilot's capabilities and see if it could fulfill my requirements.

**Comment by GhostOf0days** [comment, marked as answer]
AI having memory issues is an industry-wide issue we haven't quite fixed yet. Please see this [article](https://www.sdxcentral.com/articles/analysis/ai-has-a-long-term-memory-problem-how-to-make-neural-networks-less-forgetful/2023/03/). There is ongoing [research](https://neurosciencenews.com/biological-neural-network-memory-22729/) to try to solve this issue, however.

**Comment by JonasDev17** [comment]
That makes sense. I just noticed that Github Copilot's Memory is significantly worse than chatgpt's somehow - although I thought Github Copilot x uses gpt4 too.

**Comment by JonasDev17** [comment]
I have also switched back to the regular one now and  the regular one has much better and faster inline suggestions.

**Reply by GhostOf0days** [reply]
This could be an issue related to the model. The memory issue is likely related to capabilities of the industry as a whole. Faster may not be related to the model. If it's based on GPT-4, it should be slower to respond but give better responses due to a larger training corpus.

**Reply by GhostOf0days** [reply]
I would create a new thread and explicitly state the issue in the title.

**Comment by rochiey** [comment]
I've been using copilot since its release. I have also thought after gpt4 release, the copilot will integrate the gpt4 to itself. Looks like I am wrong? I am now currently using Github Copilot Chat Pre-release and the answer it gives is not sufficient. I am more inclined to use phind.com more than copilot. Hope they would roll copilot x soon and integrate gpt4 to it before I decide to unsubscribe since phind is already there (and its free gpt4 for 30 responses).

**Comment by Relisfoku** [comment]
disculpen me podria alguien decir como se hace un scriping ?
